server/modules/search/elasticsearch/engine.js

```diff
@@ -87,63 +84,156 @@ module.exports = {
    * @param {Object} opts Additional options
    */
   async query(q, opts) {
-    try {
-      const results = await this.client.search({
-        index: this.config.indexName,
-        body: {
-          query: {
-            bool: {
-              filter: [
-                {
-                  bool: {
-                    should: [
-                      {
-                        simple_query_string: {
-                          query: q
-                        }
-                      },
-                      {
-                        query_string: {
-                          query: "*" + q + "*"
-                        }
-                      }
-                    ],
-                    minimum_should_match: 1
+    if(q.search("date:")==0){
```

Comment:
Use `.indexOf` instead of `.search`. You're not using regex which makes it significantly slower than indexOf. Also never use `==`, always `===` unless you're specifically looking for null.